### ISSUE TITLE
expose .on and .emit after initialization

### ISSUE THREAD

**Issue by yoshuawuyts** [yoshuawuyts/barracks#21]
not recommending to do things dynamically, but being able to pass around an instance is neat.

**Comment by yoshuawuyts** [comment]
Removes the need for a large up-front object of requirements

**Comment by yoshuawuyts** [comment]
possibly leverage the `.on('foo:bar:bin:baz', () =>)` shiz; or just `.on('foo:bar', () =>)`

**Comment by yoshuawuyts** [comment]
```js
const barracks = require('barracks')

const dispatch = barracks()
dispatch.on('courses:get', () => /* do stuff */)
dispatch.on('courses:post, (next, waitFor) => waitFor(inext, next))

dispatch('courses:post', {my: 'data'}
```

**Comment by yoshuawuyts** [comment]
- remove payload delegation (only direct passing of stuff)
- remove dependance on `this`

**Comment by yoshuawuyts** [comment]
could also turn `wait` into an object with yieldable values. E.g. going the `co` route

**Comment by yoshuawuyts** [comment]
Or just make `wait` return a Promise, so it can be yielded through Babel. Mwhehhhhh

**Comment by yoshuawuyts** [comment]
Should also be infinitely nestable, like with `wayfarer`.

**Comment by yoshuawuyts** [comment]
We should probably leverage mounting semantics as outlined in https://github.com/yoshuawuyts/wayfarer/issues/12, making this the data transport counterpart to `wayfarer`.

**Comment by yoshuawuyts** [comment]
Closed in 138aeac
